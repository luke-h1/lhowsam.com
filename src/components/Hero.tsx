import Link from 'next/link';
import ButtonGroup from './ButtonGroup';
import styles from './Hero.module.scss';
import SocialList from './SocialList';

const Hero = () => {
  return (
    <section className={styles.heroContainer}>
      <span className={styles.name}>Hey ðŸ‘‹, I'm Luke</span>
      <span className={styles.description}>I'm a Software Engineer</span>
      <span className={styles.learnMore}>
        <ButtonGroup>
          <Link href="/about" className="button">
            More about me
          </Link>
          <Link href="/static/cv-latest.pdf" className="button">
            Resume
          </Link>
        </ButtonGroup>
      </span>
      <SocialList />
    </section>
  );
};

export default Hero;
