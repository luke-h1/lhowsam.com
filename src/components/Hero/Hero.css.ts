import { responsiveStyle } from '@frontend/styles/style.css';
import { variables } from '@frontend/styles/variables.css';
import { slate } from '@radix-ui/colors';
import { style } from '@vanilla-extract/css';

/* 
background: rgb(2,0,36);
background: -moz-linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(62,182,108,1) 35%, rgba(0,212,255,1) 100%);
background: -webkit-linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(62,182,108,1) 35%, rgba(0,212,255,1) 100%);
background: linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(62,182,108,1) 35%, rgba(0,212,255,1) 100%);
filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#020024",endColorstr="#00d4ff",GradientType=1);
*/

export const root = style({
  borderRadius: variables.radius.lg,
  overflow: 'hidden',
  ...responsiveStyle({
    sm: {
      marginBottom: variables.spacing.lg,
    },
  }),
  ':focus': {
    outline: 'transparent',
  },
  ':focus-visible': {
    outlineWidth: '2px',
    outlineStyle: 'solid',
    outlineOffset: '2px',
    outlineColor: variables.color.outline,
  },
});

export const overlay = style({
  backgroundImage:
    'linear-gradient(to right top, rgba(0, 0, 0 .95), rgba(0, 0, 0, .25))',
  opacity: 0.9,
  transition: 'opacity ease-in-out 0.2s',
  selectors: {
    [`${root}:hover &`]: {
      opacity: 1,
    },
  },
});

export const content = style({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-end',
  ...responsiveStyle({
    sm: {
      padding: variables.spacing.md,
    },
    md: {
      padding: variables.spacing.md,
    },
  }),
});

export const heading = style({
  color: '#fff',
  '::before': {
    fontSize: variables.fontSize.xxl,
    content: '',
    marginBottom: variables.spacing.md,
    display: 'block',
    width: '1.2rem',
    height: '1.2rem',
    backgroundColor: variables.color.highlight,
    borderRadius: variables.radius.full,
  },
});

export const description = style({
  color: slate.slate7,
  fontSize: variables.fontSize.lg,
});
