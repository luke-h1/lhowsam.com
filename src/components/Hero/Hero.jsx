/* eslint-disable react/default-props-match-prop-types */
/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import profileImage from '../../../static/luke.jpeg';
import {
  HomeHeroSection,
  GridContainer,
  IntroductionContainer,
  List,
  CollectionItem,
  HeroImage,
} from './HeroElements';

const Hero = ({ title, introduction }) => (
  <>
    <HomeHeroSection>
      <GridContainer>
        <h1>{title}</h1>
        <IntroductionContainer>
          <p>{introduction}</p>
          <hr />
          <List>
            <h3>More About Me</h3>
            <CollectionItem>
              ðŸŒ± Iâ€™m currently learning ... Javascript & React
            </CollectionItem>
            <CollectionItem>
              {' '}
              ðŸ’š I enjoying working on ... React, Testing methodologies and
              frameworks (following best practices, unit testing), Automation
              technologies (Ansible, bash ) & OOS projects
            </CollectionItem>
          </List>
        </IntroductionContainer>
        <HeroImage src={profileImage} alt="" />
      </GridContainer>
    </HomeHeroSection>
  </>
);
Hero.defaultProps = {
  title: "Hi I'm Luke ðŸ‘‹",
  introduction:
    "I'm passionate about various web technologies and making the web fast & accesible to everyone. I strive to write clean, robust & reusable code.",
};

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  introduction: PropTypes.string.isRequired,
};

export default Hero;
