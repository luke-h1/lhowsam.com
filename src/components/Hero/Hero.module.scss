.heroContainer {
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  height: 300px;
  border-radius: 15px;
  padding: 25px 0px;
}

.name {
  font-weight: 700;
  color: #fff;
  font-size: 30px;
}

.description {
  color: #fff;
  font-weight: 300;
  margin-top: 10px;
  word-break: keep-all;
  font-size: 18px;
  line-height: 1.5;
  letter-spacing: 0.2px;
}

.learnMore {
  margin-top: 20px;
  a {
    font-weight: 500;
    font-size: 12px;
    color: #fff;
  }
}
