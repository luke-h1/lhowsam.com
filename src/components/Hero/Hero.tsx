import Link from 'next/link';
import ButtonGroup from '../ButtonGroup/ButtonGroup';
import styles from './Hero.module.scss';

interface Props {
  title: string;
  description?: string;
  compact?: boolean;
}

const Hero = ({ title, compact, description }: Props) => {
  return (
    <div className={compact ? styles.wrapperCompact : styles.wrapper}>
      <h1 className={styles.title}>{title}</h1>
      {description && <p className={styles.description}>{description}</p>}
      <ButtonGroup>
        <Link href="/about" className="button">
          More about me
        </Link>
        <Link href="/static/cv-latest.pdf" className="button">
          Resume
        </Link>
      </ButtonGroup>
    </div>
  );
};
export default Hero;
