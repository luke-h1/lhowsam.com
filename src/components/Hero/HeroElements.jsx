/* eslint-disable */
import styled from 'styled-components';

export const HeroContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: center;
  background: ${(props) => props.theme.BlueBackgroundColor};
  min-height: 65vh;
  h1 { 
    font-size: clamp(1rem, 8vw, 2.7rem);

  }
  h1,p { 
    color: ${(props) => props.theme.lightTextColor};
    font-weight: 700;
  }
  p { 
    margin-top: 2rem;
    margin: 2rem 0 0 0.1rem;
    word-wrap: wrap;
    width: 50%;
    font-size: clamp(1rem, 8vw, 1.5rem);
    font-weight: bold;
    @media(max-width: 600px){
      width: 60%;
    }
  }
`;

export const IntroContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: left;
  margin-top: 2rem;
`

export const BlurbContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: left;
  align-items: left;
  max-width: 100vw;
  @media(max-width: 600px){
    margin-bottom: 1rem;
  }
`

export const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: left;
  align-items: left;
  width: 200px;
`