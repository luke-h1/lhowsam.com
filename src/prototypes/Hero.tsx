/* eslint-disable */
import { graphql, useStaticQuery } from 'gatsby';
import React from 'react';
import styled from 'styled-components';

const HeroSection = styled.div`
  padding: 0 60px;
  color: #000;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  @media (max-width: 1100px) {
    padding: 30px;
  }
`;

const HeroWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 820px;
  @media (max-width: 1100px) {
    grid-template-columns: 1fr;
  }
  @media (max-width: 468px) {
    grid-template-rows: 600px;
  }
`;

const ColumnLeft = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: flex-start;
  p {
    font-size: 16px;
    line-height: 2.3;
  }
  h1 {
    font-size: 48px;
    @media (max-width: 1100px) {
      font-size: 80px;
    }
    @media (max-width: 468px) {
      h1 {
        font-size: 32px;
      }
    }
  }
`;

const ColumnRight = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const StyledImage = styled.img`
  width: 60%;
  height: 60%;
  @media (max-width: 1100px) {
    width: 40%;
    height: 40%;
  }
`;

const Hero = () => {
  const { image } = useStaticQuery(graphql`
    query {
      image: file(relativePath: { eq: "luke.png" }) {
        sharp: childImageSharp {
          fluid {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
    }
  `);

  return (
    <>
      <HeroSection>
        <HeroWrapper>
          <ColumnLeft>
            <h1>Hey, I do xyz and lorem ipsum dolor amet stuff</h1>
            <p>
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Pariatur similique iure, fugit dolore libero delectus sint sapiente
              accusamus mollitia dolorum earum omnis odit repellat voluptatibus nostrum! Nobis blanditiis doloremque cupiditate.
            </p>
          </ColumnLeft>
          <ColumnRight>
            <StyledImage src={image.sharp.fluid.src} />
          </ColumnRight>
        </HeroWrapper>
      </HeroSection>
    </>
  );
};
export default Hero;
