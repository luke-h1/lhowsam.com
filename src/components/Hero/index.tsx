import { variables } from '@frontend/styles/variables.css';
import { Heading } from '../Heading';
import Link from '../Link';
import { Spacer } from '../Spacer';
import Text from '../Text';
import * as styles from './Hero.css';

interface Props {
  heading: string;
  description: string;
}

const Hero = ({ description, heading }: Props) => {
  return (
    <div className={styles.root}>
      <span className={styles.overlay} />
      <div className={styles.content}>
        <Heading
          fontSize={{ sm: 'xxl', md: 'xxxl' }}
          className={styles.heading}
          data-testid="highlight-heading"
        >
          {heading}
        </Heading>
        <Spacer height="sm" />
        <Text
          fontSize={{ md: 'lg' }}
          className={styles.description}
          data-testid="highlight-description"
        >
          {description}
        </Text>
        <Link href="/about" css={{ marginTop: variables.spacing.lg }}>
          More about me
        </Link>
      </div>
    </div>
  );
};
export default Hero;
