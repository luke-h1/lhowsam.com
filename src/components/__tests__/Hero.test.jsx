/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Hero from '../Hero/Hero';
import Enzyme, {shallow} from 'enzyme';
import EnzymeAdapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new EnzymeAdapter() });


test('renders component without errors', () => { 
  const wrapper = shallow(<Hero />);
  expect(wrapper).toMatchSnapshot();
})
