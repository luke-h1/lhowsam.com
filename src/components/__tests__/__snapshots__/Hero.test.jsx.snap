// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component without errors 1`] = `ShallowWrapper {}`;
